controllers/progressivesync_controller.go

```diff
@@ -388,13 +374,15 @@ func (r *ProgressiveSyncReconciler) reconcileStage(ctx context.Context, ps syncv
 	}
 	log.Info("fetched apps targeting selected clusters", "apps", utils.GetAppsName(apps))
 
+	pss.RefreshState(apps, stage)
```

Comment:
reconciling the view of the state.
goes through all the apps and figure out what's synced and what's not.